worlds/ffmq/__init__.py

```diff
@@ -0,0 +1,209 @@
+import Utils
+import settings
+import base64
+import threading
+import requests
+import yaml
+from worlds.AutoWorld import World, WebWorld
+from BaseClasses import Tutorial
+from .Regions import create_regions, location_table, set_rules, rooms, non_dead_end_crest_rooms,\
+    non_dead_end_crest_warps
+from .Items import item_table, item_groups, create_items, FFMQItem, fillers
+from .Output import generate_output
+from .Options import option_definitions
+from .Client import FFMQClient
+
+
+# removed until lists are supported
+# class FFMQSettings(settings.Group):
+#     class APIUrls(list):
+#         """A list of API URLs to get map shuffle, crest shuffle, and battlefield reward shuffle data from."""
+#     api_urls: APIUrls = [
+#         "https://api.ffmqrando.net/",
+#         "http://ffmqr.jalchavware.com:5271/"
+#     ]
+
+
+class FFMQWebWorld(WebWorld):
+    tutorials = [Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to playing Final Fantasy Mystic Quest with Archipelago.",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["Alchav"]
+    )]
+
+
+class FFMQWorld(World):
+    """Final Fantasy: Mystic Quest is a simple, humorous RPG for the Super Nintendo. You travel across four continents,
+    linked in the middle of the world by the Focus Tower, which has been locked by four magical coins. Make your way to
+    the bottom of the Focus Tower, then straight up through the top!"""
+    # -Giga Otomia
+
+    game = "Final Fantasy Mystic Quest"
+
+    item_name_to_id = {name: data.id for name, data in item_table.items() if data.id is not None}
+    location_name_to_id = location_table
+    option_definitions = option_definitions
+
+    topology_present = True
+
+    item_name_groups = item_groups
+
+    generate_output = generate_output
+    create_items = create_items
+    create_regions = create_regions
+    set_rules = set_rules
+
+    data_version = 1
+    
+    web = FFMQWebWorld()
+    # settings: FFMQSettings
+
+    def __init__(self, world, player: int):
+        self.rom_name_available_event = threading.Event()
+        self.rom_name = None
+        self.rooms = None
+        super().__init__(world, player)
+
+    def generate_early(self):
+        if self.multiworld.sky_coin_mode[self.player] == "shattered_sky_coin":
+            self.multiworld.brown_boxes[self.player] = self.multiworld.brown_boxes[self.player].from_text("include")
```

Comment:
Please change the value instead of the option directly as this breaks #993

Reply:
or would #993 break #1909?

Reply:
done